### PR TITLE
`FieldType`: Ensure Doxygen Works

### PR DESCRIPTION
**Pull request by ax3l**
Ensure `FieldType` is picked up by doxygen. Include in the manual (Sphinx) via our breathe plugin.

- [x] @WeiqunZhang it does not yet look like Doxygen can pick up `AMREX_ENUM` here. I tried adding into the `Doxyfile` another macro, but no success so far.

### CODE REVIEW

File: Source/Fields.H [hunk 1]
```diff
@@ -18,6 +18,11 @@
 
 namespace warpx::fields
 {
+    /** Unique identifiers for WarpX scalar and vector fields.
+     *
+     * These are implemented as @see amrex::MultiFab (one or one per component "direction", respectively) and stored in the
+     * @see ablastr::fields::MultiFabRegister .
+     */
     AMREX_ENUM(FieldType,
```
Comment (ax3l) [inline Source/Fields.H:26]: Maybe some hackery to `Docs/Doxyfile` can help Doxygen to pick this up...? I tried redefining `AMREX_ENUM`, but the VA_ARGS make it trickier than other macros.

Comment (ax3l) [reply Source/Fields.H:26]: Found a solution to define and expand this macro.

**Comment by ax3l** [comment]
@WeiqunZhang solved! Ready for merge :)

File: Docs/Doxyfile [hunk 2]
```diff
@@ -2305,14 +2305,16 @@ PREDEFINED             = AMREX_Linux=1 \
                          WARPX_QED=1 \
                          WARPX_QED_TABLE_GEN=1
 
+PREDEFINED           += "AMREX_ENUM(CLASS,...)=\"enum class CLASS : int { __VA_ARGS__ };\""
```
Comment (ax3l) [inline Docs/Doxyfile:2308]: Intentionally simplified.

**Comment by ax3l** [comment]
Now included in the manual. This renders now in my local Sphinx
image
- doxygen 1.9.7
- sphinx 7.1.2
- breathe 4.35.0

**Comment by ax3l** [comment]
Preview works now as well (after Doxygen update on RTD to a newer version than 1.9.1, which comes with Ubuntu 22.04 by default):
https://warpx--5330.org.readthedocs.build/en/5330/developers/fields.html#field-names

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**